### PR TITLE
Feature: Startup Activities

### PR DESCRIPTION
**Pull request by Andale-online**
Hey dude...
Had some troubles with me fork... Had to delete a lot.

Here's the Startup Activity #203 

Greetings

### CODE REVIEW

File: src/activities/Elsa.Activities.Startup/Elsa.Activities.Startup.csproj [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <PackageVersion>1.0.0</PackageVersion>
+        <LangVersion>latest</LangVersion>
+        <Authors>Elsa Contributors</Authors>
+        <Description>
+            Elsa is a set of workflow libraries and tools that enable super-fast workflowing capabilities in any .NET Core application.
+            This package provides the following Console activities:
+            
+            * ReadLine
+            * WriteLine
```
Comment (sfmskywalker) [inline src/activities/Elsa.Activities.Startup/Elsa.Activities.Startup.csproj]: Maybe replace these with:

```markdown
* Startup
```

File: src/activities/Elsa.Activities.Startup/Activities/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    {
+
+        public Startup()
+        {
+        }
+
+
+        protected override ActivityExecutionResult OnExecute(WorkflowExecutionContext context)
+        {
+            return Halt();
```
Comment (sfmskywalker) [inline src/activities/Elsa.Activities.Startup/Activities/Startup.cs]: I'm not sure I understand how this works. It looks like that the activity will halt, but I don't see at which point the activity will be resumed. The `StartupHostedService` will trigger workflows starting with the `Startup` activity, which will cause the `OnExecute` method to be invoked. But then the workflow becomes suspended (the new term for "halted" in Eldsa 2.0).

Unless I'm missing something?

File: src/activities/Elsa.Activities.Startup/Elsa.Activities.Startup.csproj (hunk 1 above)
Comment (Andale-online) [reply src/activities/Elsa.Activities.Startup/Elsa.Activities.Startup.csproj]: Sorry, of course

File: src/activities/Elsa.Activities.Startup/Activities/Startup.cs (hunk 2 above)
Comment (Andale-online) [reply src/activities/Elsa.Activities.Startup/Activities/Startup.cs]: Dunno, copied from timer activities. It works :-D
Shall I do onexecute instead?

Comment (sfmskywalker) [reply src/activities/Elsa.Activities.Startup/Activities/Startup.cs]: I see. So here's the thing: the Timer activities work because they are repeatedly being triggered. Internally, thee activities compare the current time with the first time they got triggered. If the difference exceeds a given threshold, the activity executes and continues the workflow.

The Startup background task on the other hand only triggers the Startup activity once. Since the Startup activity returns `HaltResult`, it will cause the workflow to be suspended. Only when you kill the ost and restart it, will the workflow continue. 

At least, this is the behavior I'd expect to see by looking at the code.

To summarize: please do return `Done()` from `OnExecute` :)

File: src/activities/Elsa.Activities.Startup/Elsa.Activities.Startup.csproj (hunk 1 above)
Comment (Andale-online) [reply src/activities/Elsa.Activities.Startup/Elsa.Activities.Startup.csproj]: Fixed

File: src/activities/Elsa.Activities.Startup/Activities/Startup.cs (hunk 2 above)
Comment (Andale-online) [reply src/activities/Elsa.Activities.Startup/Activities/Startup.cs]: Fixed :-)

**Comment by Andale-online** [comment]
Give it a try
